src/Network/XMPP/Types.hs

```diff
@@ -272,8 +337,8 @@ data Stanza :: StanzaType -> * where
         }
         -> Stanza 'Message
     MkPresence ::
-        { pFrom     :: Maybe (JID '[])
-        , pTo       :: Maybe (JID '[])
+        { pFrom     :: Maybe SomeJID
```

Comment:
Same as above, we have PresenceType which could become Stanza's phantom and used with two separate type families to calculate correct from and to JIDs.